wagtail/core/models/__init__.py

```diff
@@ -248,6 +237,25 @@ def __str__(self):
         return force_str(self.get_display_name() or self.language_code)
 
 
+@receiver(pre_validate_delete, sender=Locale)
```

Comment:
question: should this not live in `core.signal_handlers`? or this is too edge case for that?

Reply:
I debated that a bit... there's precedent for putting signal handlers here (`set_locale_on_new_instance`, now in models/i18n.py) and intuitively it does feel like these are part of the model definition in a way that more conventional uses of signals (e.g. sending a notification) wouldn't be. The Django docs [do advise against putting signal handlers in models.py](https://docs.djangoproject.com/en/3.2/topics/signals/#connecting-receiver-functions) (see under "Where should this code live?"), but I don't entirely follow the reasoning there, and in this case it wouldn't be a problem if it did end up being registered multiple times. I don't have a particularly strong feeling either way, and I'd be happy to move it to signal_handlers if that's the consensus.